model/profile.go

```diff
@@ -66,6 +67,11 @@ func (pp PprofProfile) Transform(ctx context.Context, _ *transform.Config) []bea
 		profileID = fmt.Sprintf("%x", uuid)
 	}
 
+	// Profiles are stored in their own "metrics" data stream, with a data
+	// set per service. This enables managing retention of profiling data
+	// per-service, and indepedently of lower volume metrics.
+	dataset := fmt.Sprintf("%s.profiling", datastreams.NormalizeServiceName(pp.Metadata.Service.Name))
```

Comment:
any chance we can do this `datastreams.NormalizeServiceName` call only once for all the events?

Reply:
It would be possible by having each event define a dataset prefix, and then defining a script processor. It will be more convoluted, which is why I went this way.

I haven't thought of any other better options - any ideas?

Reply:
Not now, was just curious